compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -402,7 +402,17 @@ OMR::ARM64::TreeEvaluator::aloadEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    node->setRegister(tempReg);
 
    TR::MemoryReference *tempMR = new (cg->trHeapMemory()) TR::MemoryReference(node, 8, cg);
-   generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, tempReg, tempMR);
+
+   if (TR::Compiler->om.generateCompressedObjectHeaders() &&
+       (node->getSymbol()->isClassObject() ||
+        (node->getSymbolReference() == comp->getSymRefTab()->findVftSymbolRef())))
+      {
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmw, node, tempReg, tempMR);
+      }
+   else
+      {
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, tempReg, tempMR);
+      }
```

Comment:
To save some code space, can you just set the opcode in the if/else and then move the `generateTrgMemInstruction` afterwards?

Reply:
Changed the code as suggested.